common/modules/farm/farm_position/src/farm_position.rs

```diff
@@ -0,0 +1,105 @@
+#![no_std]
+
+multiversx_sc::imports!();
+multiversx_sc::derive_imports!();
+
+use common_structs::FarmTokenAttributes;
+
+#[multiversx_sc::module]
+pub trait FarmPositionModule:
+    farm_token::FarmTokenModule
+    + utils::UtilsModule
+    + permissions_module::PermissionsModule
+    + multiversx_sc_modules::default_issue_callbacks::DefaultIssueCallbacksModule
+{
+    #[payable("*")]
+    #[endpoint(updateTotalFarmPosition)]
+    fn update_total_farm_position(&self) {
+        let caller = self.blockchain().get_caller();
+        let payments = self.get_non_empty_payments();
+        let farm_token_mapper = self.farm_token();
+        let farm_token_id = farm_token_mapper.get_token_id();
+        let mut new_total_farm_position = BigUint::zero();
+        for farm_position in &payments {
+            require!(
+                farm_position.token_identifier == farm_token_id,
+                "Bad payment token"
+            );
+            let token_attributes: FarmTokenAttributes<Self::Api> =
+                farm_token_mapper.get_token_attributes(farm_position.token_nonce);
+
+            if token_attributes.original_owner != caller {
+                self.user_total_farm_position(&token_attributes.original_owner)
+                    .update(|user_farm_position| {
+                        if *user_farm_position > farm_position.amount {
+                            *user_farm_position -= &farm_position.amount;
+                        } else {
+                            *user_farm_position = BigUint::zero();
+                        }
+                    });
+            }
+
+            new_total_farm_position += farm_position.amount;
+        }
+
+        let user_current_farm_position = self.user_total_farm_position(&caller).get();
+        if new_total_farm_position > user_current_farm_position {
+            self.user_total_farm_position(&caller)
+                .set(new_total_farm_position)
+        }
+
+        self.send().direct_multi(&caller, &payments);
+    }
+
+    fn check_and_update_user_farm_position(
```

Comment:
this is called outside in a form of for payment range payments -> check and update. Introduce that for inside, and compute the total_farm_position from all those, to make the update.

I think the code of if user.total_farm_position == 0 -> set user farm position is missing. That feature could be added here, as this check_and_update is called everywhere.

Reply:
Updated.